tests/integration_tests.py

```diff
@@ -196,3 +197,50 @@ def test_create_job_with_upload(local_grand_challenge, files):
     assert len(job["inputs"]) == 1
     job = c.algorithm_jobs.detail(job["pk"])
     assert job["status"] == "Queued"
+
+
+def test_get_algorithm_by_slug(local_grand_challenge):
+    c = Client(
+        base_url=local_grand_challenge,
+        verify=False,
+        token=DEMO_PARTICIPANT_TOKEN,
+    )
+
+    by_slug = c.algorithms.detail(slug="test-algorithm-evaluation-1")
+    by_pk = c.algorithms.detail(pk=by_slug["pk"])
+
+    assert by_pk == by_slug
+
+
+def test_get_reader_study_by_slug(local_grand_challenge):
+    c = Client(
+        base_url=local_grand_challenge, verify=False, token=READERSTUDY_TOKEN,
+    )
+
+    by_slug = c.reader_studies.detail(slug="reader-study")
+    by_pk = c.reader_studies.detail(pk=by_slug["pk"])
+
+    assert by_pk == by_slug
+
+
+@pytest.mark.parametrize("key", ["slug", "pk"])
+def test_detail_no_objects(local_grand_challenge, key):
+    c = Client(
+        base_url=local_grand_challenge, verify=False, token=READERSTUDY_TOKEN,
+    )
+
+    with pytest.raises(ObjectNotFound):
+        c.reader_studies.detail(**{key: "foo"})
+
+
+def test_detail_multiple_objects(local_grand_challenge):
+    c = Client(token=ADMIN_TOKEN, base_url=local_grand_challenge, verify=False)
+
+    file_to_upload = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)), "testdata", "rnddata"
```

Comment:
Random find: I am pretty sure that abspath is to preferred over realpath in pretty much all cases. This mainly has to do with how libraries - like this snippet of code - are resolving the place where data is loaded. If someone externally decides to link this single test file into some other environment with a symbolic, realpath will resolve that symbolic link and point to the original path even though that is not the file at the location where the user invoked it.

If one uses abspath instead, you also get a canonical representation of the path that includes the symbolic links, and therefore still points in the environment that the uses was calling the file from. Python internally clearly prefers abspath as it ignores any kind of symbolic links in its library paths it seems. Node-js falls apart when there is a symbolic links in any path because library developers seem quite confused with symbolic links and like to mix abspath with realpath :P

Anyway, random find. I think this does not need to be a blocker and can be fixed when it becomes a problem.

Reply:
Yeah, that was [copy-pasted from a test above](https://twitter.com/kelseyhightower/status/1022837097743319040). Test code so not worried about it.